File: 70.js
// 70. Climbing Stairs
// You are climbing a staircase. It takes n steps to reach the top.
// Each time you can either climb 1 or 2 steps. In how many distinct ways can you climb to the top?

var climbStairs = function(n) {
    let arr = [0, 1]
    for (let i = 1; i < n; i++) {
        arr.push(arr[i] + arr[i - 1])
    }
    return arr[arr.length - 1] + arr[arr.length - 2]
}

console.log(climbStairs(3))

File: 70.py
# 70. Climbing Stairs
# You are climbing a staircase. It takes n steps to reach the top.
# Each time you can either climb 1 or 2 steps. In how many distinct ways can you climb to the top?

class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 1:
            return 1
        ways = [1, 2]
        for _ in range(n - 2):
            next = ways[0] + ways[1]
            ways[0] = ways[1]
            ways[1] = next
        return ways[1]

s = Solution()
print(s.climbStairs(2))
print(s.climbStairs(3))
print(s.climbStairs(10))